pip/req/req_set.py

```diff
@@ -123,6 +133,68 @@ def prep_for_dist(self):
         self.req_to_install.assert_source_matches_version()
 
 
+class SetupCfgDistribution(pkg_resources.Distribution):
+
+    def __init__(self, cfg, location=None):
+        self._cfg = cfg
+        project_name = cfg.get('metadata', 'name')
+        super(SetupCfgDistribution, self).__init__(
+            location=location, project_name=project_name)
+
+    @property
+    def _dep_map(self):
+        try:
+            return self.__dep_map
+        except AttributeError:
+            # requires
+            try:
+                requires = self._cfg.get('metadata', 'requires-dist')
+            except configparser.NoOptionError:
```

Comment:
Thoughts on making private helper methods, e.g.,

```py
try:
    return self.__dep_map
except AttributeError:
    requires = self._get_requires()
    extras = self._get_extras()
    return self._build_dep_map(requires, extras)
```

Also, does it make sense to cache the built dependency map in `self.__dep_map` so future calls to this just return immediately instead of re-parsing each time?